test/jdk/java/net/SocketOption/OptionsTest.java

```diff
@@ -107,7 +107,12 @@ static NetworkInterface getNetworkInterface() {
             while (nifs.hasMoreElements()) {
                 NetworkInterface ni = nifs.nextElement();
                 if (ni.supportsMulticast()) {
-                    return ni;
+                    try (MulticastSocket ms = new MulticastSocket()) {
+                        ms.setNetworkInterface(ni);
+                        return ni;
+                    } catch (SocketException ex) {
+                        // continue
+                    }
```

Comment:
isn't that going to hide bugs that the test would have detected?

Reply:
I don't think so; if all interfaces are skipped, the test will fail - setting a null interface is not permitted.

Other multicast-related tests do similar things:
- `java/net/MulticastSocket/TestInterfaces.java` [skips interfaces that throw an exception](https://github.com/openjdk/jdk/blob/master/test/jdk/java/net/MulticastSocket/TestInterfaces.java#L103)
- `NetworkConfiguration.multicastInterfaces` [skips interfaces that don't have a network address](https://github.com/openjdk/jdk/blob/master/test/lib/jdk/test/lib/NetworkConfiguration.java#L198); we can't do that here because we're testing with a security manager and can't retrieve addresses.

Reply:
Could you try something like to find an interface with an IP address configured:
```
        boolean preferIPv4 = Boolean.getBoolean("java.net.preferIPv4Stack");
        Class<? extends InetAddress> clazz = preferIPv4 ? Inet4Address.class : Inet6Address.class;
        var ni = NetworkInterface.networkInterfaces()
                .filter(_ni -> _ni.inetAddresses().anyMatch(clazz::isInstance))
                .findAny()
                .orElse(null);
```

Reply:
Yeah, I had something like that before. With security manager installed it will only return the loopback interface, which is not usable for multicasting on Linux.

The test is using [this SecurityManager policy](https://github.com/openjdk/jdk/blob/master/test/jdk/java/net/SocketOption/options.policy), which only allows `connect` to `localhost`, so only loopback address will be visible on NetworkInterfaces because of [this check](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/net/NetworkInterface.java#L143)

Reply:
OK then. I updated the SM policy to allow all interface addresses to be retrieved, and updated the test to only consider interfaces with at least one address.
The test is never run with `preferIPv4Stack`, and even if it were, `preferIPv4Stack` would filter out all IPv6 interface addresses.
The only way this test could spuriously fail now is if it got an interface with IPv4 addresses only. Which is possible, but then `test/jdk/java/net/MulticastSocket/SetGetNetworkInterfaceTest.java` would also fail.